### PR TITLE
fix(event-search): make search case insensitive for postgres

### PR DESCRIPTION
**Pull request by hazzadous**
This change is in reference to https://github.com/PostHog/posthog/issues/5812

At the moment there are two paths to querying event definitions:

 1. clickhouse path in `ee/`
 2. postgres path in `posthog/`

The search is already case insensitive in clickhouse so we just needed to update to use icontains as opposed to contains for the postgres path.

I've also refactored the test a little to not be quite so heavy, and be more localised on its configuration.

One thing I'm not sure about is why we'd not use postgres tsv here instead of doing our own tokenising etc.

### CODE REVIEW

File: posthog/api/test/test_event_definition.py [hunk 1]
```diff
@@ -118,3 +142,63 @@ def test_query_event_definitions(self):
...
+
+def create_organization(name: str) -> Organization:
+    return Organization.objects.create(name=name)
+
+
+def create_user(email: str, password: str, organization: Organization):
+    return User.objects.create_and_join(organization, email, password)
+
+
+def create_team(organization: Organization) -> Team:
```
Comment (EDsCODE) [inline posthog/api/test/test_event_definition.py:155]: Followup: could consider moving this to some test utility directory and reuse across other postgres tests

**Review by EDsCODE — APPROVED** [review]
lgtm

File: posthog/api/test/test_event_definition.py (hunk 1 above)
Comment (hazzadous) [reply posthog/api/test/test_event_definition.py:155]: Yes I’m thinking we either have corresponding tests for endpoints or admin actions that are responsible for these actions, in which we can define these canonical helper functions

**Comment by hazzadous** [comment]
> lgtm

Should I be concerned about the saml test failure?

**Comment by EDsCODE** [comment]
nope SAML test failure should be unrelated

**Pull request merged**